ui/src/app/pages/admin/admin-egress-audit.tsx

```diff
@@ -0,0 +1,170 @@
+};
+
+const HighlightedLogMessage = ({logPattern, msg}: {logPattern: string, msg: string}) => {
+  const target = logPattern.replace(/%/g, '');
+  const parts = msg.split(target);
+  return <>
+    {parts.map((part, i) => <span key={i}>
+      {part}
+      {i < parts.length - 1 ? <span style={{fontWeight: 600}}>{target}</span> : null}
+    </span>)}
+  </>;
+}
+
+export const AdminEgressAudit = (props: WithSpinnerOverlayProps) => {
+  const {eventId = ''} = useParams<MatchParams>();
+  const [egressDetails, setEgressDetails] = useState<AuditEgressEventResponse>(null);
+  const [pendingUpdateEvent, setPendingUpdateEvent] = useState<EgressEvent>(null);
+  const [activeLogGroup, setActiveLogGroup] = useState<number>(0);
+
+  useEffect(() => {
+    const aborter = new AbortController();
+    (async() => {
+      const resp = await egressEventsAdminApi().auditEgressEvent(eventId, {}, {
+        signal: aborter.signal
+      });
+      setEgressDetails(resp);
+      props.hideSpinner();
+    })();
+    return () => aborter.abort();
+  }, []);
+
+  const onSave = useCallback(async() => {
+    if (!pendingUpdateEvent) {
+      return;
+    }
+
+    if (fp.isEqual(egressDetails.egressEvent, pendingUpdateEvent)) {
+      return;
+    }
+
+    props.showSpinner();
+    const updatedEvent = await egressEventsAdminApi().updateEgressEvent(
+      pendingUpdateEvent.egressEventId, {egressEvent: pendingUpdateEvent});
+    setEgressDetails({
+      ...egressDetails,
+      egressEvent: updatedEvent
+    });
+    setPendingUpdateEvent(null);
+    props.hideSpinner();
+  }, [egressDetails, pendingUpdateEvent]);
+
+  if (!egressDetails) {
+    return null;
+  }
+
+  const event = egressDetails.egressEvent;
+  const [username,] = event.sourceUserEmail.split('@');
+  return <div style={{padding: '0 20px'}}>
+    <h2>Egress event {event.egressEventId}</h2>
+    <div style={{display: 'table', marginBottom: '15px'}}>
+      <DetailRow label="Detection time">
+        {(new Date(event.creationTime)).toString()}
+      </DetailRow>
+      <DetailRow label="Source user">
+        <StyledRouterLink path={`/admin/users/${username}`}>
+          {event.sourceUserEmail}
+        </StyledRouterLink>
+      </DetailRow>
+      <DetailRow label="Source workspace">
+        <StyledRouterLink path={`/admin/workspaces/${event.sourceWorkspaceNamespace}`}>
+          {event.sourceWorkspaceNamespace}
+        </StyledRouterLink>
+      </DetailRow>
+      <DetailRow label="Google project">
+        {event.sourceGoogleProject}
+      </DetailRow>
+      <DetailRow label="Egress volume">
+        {event.egressMegabytes?.toFixed(2)} MB (over {event.egressWindowSeconds / 60} min)
+      </DetailRow>
+      <DetailRow label="Status">
+        <Dropdown
+          value={pendingUpdateEvent?.status ?? event.status}
+          options={mutableEgressEventStatuses}
+          onChange={(e)=> {
+            setPendingUpdateEvent({
+              ...event,
+              status: e.value
```

Comment:
Shall we compare 
`egressDetails.status !== e.value?` here or in the button enabled below? 
i.e. we block user updating egress event if there is no diff

Reply:
Good catch, added.